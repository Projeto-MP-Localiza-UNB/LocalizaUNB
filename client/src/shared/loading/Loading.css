.loading-container {
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;
  z-index: 5;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.4);
  backdrop-filter: blur(1px);
  color: white;
  font-family: Inter;
  font-size: 1.5rem;
  text-align: center;
}
