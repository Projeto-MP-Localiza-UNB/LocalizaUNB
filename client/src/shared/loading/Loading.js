import loading from './assets/Ripple-1s-200px.svg';
import './Loading.css';

export default function Loading({ status, message = '' }) {
  return (
    <>
      {status ? (
        <div className="loading-container">
          <img src={loading} alt="Ícone de carregamento" />
          <p>{message}</p>
        </div>
      ) : (
        <></>
      )}
    </>
  );
}
