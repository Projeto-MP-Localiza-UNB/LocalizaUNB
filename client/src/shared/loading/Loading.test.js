import renderer from 'react-test-renderer';
import Loading from './Loading';

it('Renders the loading symbol correctly after status change', () => {
  let component = renderer.create(<Loading status={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(<Loading status={true} />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders the loading symbol with given message correctly', () => {
  const component = renderer.create(
    <Loading status={true} message="Lorem Ipsum" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
