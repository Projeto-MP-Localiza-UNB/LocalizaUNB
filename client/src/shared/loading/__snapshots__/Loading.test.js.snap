// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the loading symbol correctly after status change 1`] = `null`;

exports[`Renders the loading symbol correctly after status change 2`] = `
<div
  className="loading-container"
>
  <img
    alt="Ícone de carregamento"
    src="Ripple-1s-200px.svg"
  />
  <p />
</div>
`;

exports[`Renders the loading symbol with given message correctly 1`] = `
<div
  className="loading-container"
>
  <img
    alt="Ícone de carregamento"
    src="Ripple-1s-200px.svg"
  />
  <p>
    Lorem Ipsum
  </p>
</div>
`;
